src/toolkit/solana-errors.ts

```diff
@@ -0,0 +1,55 @@
+
+export interface SolanaError {
+    code: number;
+    message: string;
+}
+
+export const SOLANA_ERRORS: Record<number, string> = {
+    1: 'insufficient solana or token balance',
+    
+    1001: 'the account you are trying to spend from does not have enough SOL or has not been properly initialized. A newly generated account might require an initial airdrop of SOL or a specific transaction to initialize its state before it can store funds. ',
+
+    2001: 'The specified position account was not found. It may have been closed, expired, or the account address is incorrect.',
+
+    3001: 'Transaction has expired due to network congestion. Please try submitting the transaction again.',
+    3007: 'The given account is owned by a different program than expected.',
+    
+    4001: 'The string to be decoded is not correctly encoded.',
+
+    6024: 'You might be passing an amount or min_sol_amount that is too large and is resulting in an overflow when calculating the amount of tokens that you will receive.',
+
+};
+
+export function getSolanaErrorInfo(error: any): {
+    code: number;
+    message: string;
+} {
+    const errorText = error.message || error.toString();
+    if (errorText.includes('custom program error: 0x1')) {
```

Comment:
this will match `custom program error: 0x1234`

Reply:
yes, modified.